src/AWS.Deploy.Orchestration/DeploymentBundleHandler.cs

```diff
@@ -267,5 +274,43 @@ private async Task PushDockerImage(string targetTagName)
                 throw new DockerPushFailedException(DeployToolErrorCode.DockerPushFailed, errorMessage, result.ExitCode);
             }
         }
+
+        /// <summary>
+        /// Inspects the already built docker image by using 'docker inspect'
+        /// to return the environment variables used in the container.
+        /// </summary>
+        /// <param name="recommendation">The currently selected recommendation</param>
+        /// <param name="sourceTag">The docker tag of the built image</param>
+        /// <returns>A dictionary that represents the environment varibales from the container</returns>
+        public async Task<Dictionary<string, string>> InspectDockerImageEnvironmentVariables(Recommendation recommendation, string sourceTag)
+        {
+            var dockerInspectCommand = "docker inspect --format \"{{ index (index .Config.Env) }}\" " + sourceTag;
+            var result = await _commandLineWrapper.TryRunWithResult(dockerInspectCommand, streamOutputToInteractiveService: false);
+
+            if (result.ExitCode != 0)
```

Comment:
Since this is being done for warning purposes should we fail the build if this fails or should we put out a warning that we failed to perform or check?

Reply:
I'm fine skipping the warning, maybe add as a verbose logging if we need to diagnose anything. 

But sense you are throwing the `DockerInspectFailedException` exception if the inspect fails won't that break the build? Seems like we need to rework that.

Reply:
I added a debug log so that it shows up when we turn on diagnostics.

As for the `DockerInspectFailedException`, that follows the pattern of docker build, tag and push. We are separating concerns here, so if the docker inspect fails, we'll throw an exception that it failed. Though up in call stack, we are catching that exception and providing the appropriate messaging. This is done in the orchestrator.

Reply:
Ok